packages/react-components/src/components/SendBox.tsx

```diff
@@ -65,6 +65,11 @@ export interface ActiveFileUpload {
    * Error message to be displayed to the user if the upload fails.
    */
   errorMessage?: string;
+
+  /**
+   * `true` means that the upload is completed.
+   */
+  isUploaded?: boolean;
```

Comment:
is isUploaded the same as progress having value of 1?

Reply:
Yes and metadata should be populated

Reply:
We can't rely on just the upload progress, since it can be 1 without metadata being populated.
In our case, metadata being populated marks an upload as complete irrespective of it's upload progress

Reply:
Not able to understand why progress is not reliable, isn't it updated to 1 when file upload is complete.

Reply:
Naming nit: `uploadComplete` is more assertive that something completed.

I agree that it's better to have this prop as separate from `progress`. Progress is used just for the progress bar. Through the file sharing API, the two are sperate (different API to notify file upload completion, separate state etc.)

Reply:
It is but it doesn't guarantee that metadata is populated.

Reply:
progress is controlled by Contoso's logic, so there is no way for us to ensure that progress of 1 equals metadata being populated.

Reply:
I guess the confusion is that `progress === 1` does not actually mean the file upload progress has truly completed - there is an extra processing step (performed by the server/place it is uploaded to?) where metadata is populated and added to the file.
If I've understood that correctly I would suggest either marking this property not as optional but as required and rename to something like `readyToSend`, or have two interfaces:
`ActiveFileUploads` and `CompletedFileUploads`.

The reason I would opt for two different interfaces is I can set the progress to 50% here, but also set uploadComplete as true which doesn't make sense

Reply:
I think making `isUploaded` a required prop might be a good start.
Multiple interfaces will add complexity.

Reply:
Sounds good, maybe just add a comment in the tsdoc comment to progress mentioning how their related.
```
/** @remarks if isUploaded is true, progress bar will not be shown */
progress
```